### PR TITLE
Enable additional lints

### PR DESCRIPTION
**Pull request by jamescurtin**
This is the last significant linting PR that I plan to make. I have a few other ideas for where I might turn next, but will start discussion in an issue.

The first commit introduces a series of additional linting rules that, aspirationally, would be enabled for the project. Rules that passed out of the gate are enabled and the remainder are commented out. Each subsequent commit enables an additional rule and corresponding code changes. There are still a number of rules that are still commented out, but I believe these can be enabled in the future. There were two reasons why I didn't go for enabling all of them now:

1. Some of the lints are easy to enable, but other work should be done before turning them on. Examples include `flake8-print` (disallows print statements; I would like to introduce a logging framework and convert them to logs instead) and `eradicate` (disallows large blocks of commented out code; I don't (yet) have a good enough sense to know what's dead code that should be deleted vs. something else)
1. Some of the lints would be hard to enable right now because resolving them requires making real code changes (more than the superficial changes I've been making so far). 

While some of these lint rules were auto-fixable, the majority were hand edits. If there are lint rules that you either (a) aren't sure are worth enforcing, or (b) seems too risky at this point in time without tests to back us up, let me know and I can drop the corresponding commit for now.

### CODE REVIEW

File: auctions/consumers.py [hunk 1]
```diff
@@ -38,9 +38,8 @@ def check_bidding_permissions(lot, user):
         ).first()
         if not tos:
             return "You haven't joined this auction"
-        else:
-            if not tos.bidding_allowed:
-                return "This auction requires admin approval before you can bid"
+        if not tos.bidding_allowed:
```
Comment (iragm) [inline auctions/consumers.py:41]: this won't work, will it?  tos can be None

**Comment by iragm** [comment]
This looks good (the tests are so much more readable now!) but I am scared about making lots of changes to consumers.py, it's been years since I made major changes to that code there are either few or no tests at all.

Can we pull flake8-return and simplify for now?  They are the main ones that mess with consumers.py:
[752e2aa](https://github.com/iragm/fishauctions/pull/203/commits/752e2aac158c5c99df51b151e68bc5fb70672779)
[30d1e4f](https://github.com/iragm/fishauctions/pull/203/commits/30d1e4f0744d4fc5a3fbea91d0902dc2bcd6e60d)

I agree that long term they are good, but I think we need more tests first.

A proper logging framework would be great, if you make a PR that introduces it I will gladly start using it.

File: auctions/consumers.py (hunk 1 above)
Comment (jamescurtin) [reply auctions/consumers.py:41]: I'm not sure if there's a possible issue here in terms of the correct logic, but it should be functionally equivalent to what was happening before.

Reason being: nesting the second `if` statement under an `else` is duplicative when the first `if` contains a `return` (the second `if` statement is evaluated under the same circumstances regardless of if it's nested under the `else` or not.

**Comment by jamescurtin** [comment]
> Can we pull flake8-return and simplify for now? They are the main ones that mess with consumers.py: [752e2aa](https://github.com/iragm/fishauctions/pull/203/commits/752e2aac158c5c99df51b151e68bc5fb70672779) [30d1e4f](https://github.com/iragm/fishauctions/pull/203/commits/30d1e4f0744d4fc5a3fbea91d0902dc2bcd6e60d)

All set!

**Pull request merged**